tasks/go_deps.py

```diff
@@ -115,3 +115,110 @@ def send_count_metrics(
         print(color_message("Sending metrics to Datadog", "blue"))
         send_metrics(series=series)
         print(color_message("Done", "green"))
+
+
+BINARY_TO_TEST = ["serverless"]
+
+
+@task
+def test_dependencies_list(
+    ctx: Context,
+):
+    """
+    Compare the dependencies list for the binaries in BINARY_TO_TEST with the actual dependencies of the binaries.
+    If the lists do not match, the task will raise an error.
+    """
+    missmatch_binaries = set()
+
+    for binary in BINARY_TO_TEST:
+        binary_info = BINARIES[binary]
+        entrypoint = binary_info["entrypoint"]
+        platforms = binary_info["platforms"]
+        flavor = binary_info.get("flavor", AgentFlavor.base)
+        build = binary_info.get("build", binary)
+
+        with ctx.cd(entrypoint):
+            for platform in platforms:
+                platform, arch = platform.split("/")
+
+                goos, goarch = GOOS_MAPPING[platform], GOARCH_MAPPING[arch]
+
+                filename = os.path.join(ctx.cwd, f"dependencies_{goos}_{goarch}.txt")
+                if not os.path.isfile(filename):
+                    print(
+                        f"File {filename} does not exist. To execute the dependencies list check for the {binary} binary, please run the task `inv -e go-deps.dependencies-generate --binaries {binary}"
+                    )
+                    continue
+
+                deps_file = open(filename)
+                deps = deps_file.read()
+                deps_file.close()
+
+                build_tags = get_default_build_tags(build=build, flavor=flavor, platform=platform)
+
+                env = {"GOOS": goos, "GOARCH": goarch, "CGO_ENABLED": "1"}
+                cmd = "go list -f '{{ join .Deps \"\\n\"}}'"
+
+                res = ctx.run(
+                    f"{cmd} -tags {','.join(build_tags)}",
+                    env=env,
+                    hide='out',  # don't hide errors
+                )
+                assert res
+
+                if res.stdout.strip() != deps:
+                    missmatch_binaries.add(binary)
+
+    if len(missmatch_binaries) > 0:
+        raise Exit(
+            code=1,
+            message=color_message(
+                f"Dependencies list for {list(missmatch_binaries)} does not match. To fix this check, please run `inv -e go-deps.dependencies-generate --binaries {','.join(missmatch_binaries)}`",
+                "red",
+            ),
+        )
+
+
+@task
+def dependencies_generate(
```

Comment:
The namespace is `go-deps`, so maybe just `generate` ? (it would be used as `inv go-deps.dependencies_generate`)